### PR TITLE
python39Packages.cliff: fix missing dependency, tests and adopt to openstack team

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137557` at 6bb1eb1c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/749a7a7caa8058cd277ffbafef4870c6">openstackclient</a> (python39Packages.python-openstackclient)</li>
    <li>python38Packages.openstacksdk</li>
    <li><a href="https://gist.github.com/148ffa8c3708ba9daac108968b364cf9">python38Packages.osc-lib</a></li>
    <li><a href="https://gist.github.com/98420ca33971581e27fe2b9881fff4dd">python38Packages.python-openstackclient</a></li>
    <li>python39Packages.openstacksdk</li>
    <li><a href="https://gist.github.com/b20a1baedd174dbf88fa48cb92a5be04">python39Packages.osc-lib</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.optuna</li>
    <li>python39Packages.optuna</li>
  </ul>
</details>
<details>
  <summary>34 packages built successfully:</summary>
  <ul>
    <li>python38Packages.autopage</li>
    <li>python38Packages.cliff</li>
    <li>python38Packages.hacking</li>
    <li>python38Packages.keystoneauth1</li>
    <li>python38Packages.oslo-concurrency</li>
    <li>python38Packages.oslo-config</li>
    <li>python38Packages.oslo-context</li>
    <li>python38Packages.oslo-db</li>
    <li>python38Packages.oslo-i18n</li>
    <li>python38Packages.oslo-log</li>
    <li>python38Packages.oslo-serialization</li>
    <li>python38Packages.oslo-utils</li>
    <li>python38Packages.python-cinderclient</li>
    <li>python38Packages.python-keystoneclient</li>
    <li>python38Packages.python-novaclient</li>
    <li>python38Packages.stestr</li>
    <li>python38Packages.subunit2sql</li>
    <li>python39Packages.autopage</li>
    <li>python39Packages.cliff</li>
    <li>python39Packages.hacking</li>
    <li>python39Packages.keystoneauth1</li>
    <li>python39Packages.oslo-concurrency</li>
    <li>python39Packages.oslo-config</li>
    <li>python39Packages.oslo-context</li>
    <li>python39Packages.oslo-db</li>
    <li>python39Packages.oslo-i18n</li>
    <li>python39Packages.oslo-log</li>
    <li>python39Packages.oslo-serialization</li>
    <li>python39Packages.oslo-utils</li>
    <li>python39Packages.python-cinderclient</li>
    <li>python39Packages.python-keystoneclient</li>
    <li>python39Packages.python-novaclient</li>
    <li>python39Packages.stestr</li>
    <li>python39Packages.subunit2sql</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/cliff/default.nix:47:0:
```
   |
47 |     description = "Command Line Interface Formulation Framework";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/python-modules/autopage/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "autopage";
+  version = "0.4.0";
+
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18f511d8ef2e4d3cc22a986d345eab0e03f95b9fa80b74ca63b7fb001551dc42";
+  };
+
```
Comment (superherointj) [inline pkgs/development/python-modules/autopage/default.nix:13]:
```
  checkInputs = [
    fixtures
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/autopage/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi }:
```
Comment (superherointj) [inline pkgs/development/python-modules/autopage/default.nix:1]: `, pytestCheckHook, fixtures`

File: pkgs/development/python-modules/cliff/default.nix [hunk 3]
```diff
@@ -23,34 +20,33 @@ buildPythonPackage rec {
...
-  pytestFlagsArray = [
-    "cliff/tests/test_utils.py"
-    "cliff/tests/test_app.py"
-    "cliff/tests/test_command.py"
-    "cliff/tests/test_help.py"
-    "cliff/tests/test_lister.py"
-  ];
+  passthru.tests = {
+    pytest = callPackage ./tests.nix {};
+  };
```
Comment (superherointj) [inline pkgs/development/python-modules/cliff/default.nix:45]: Add: ` pythonImportsCheck = [ "cliff" ]; `

File: pkgs/development/python-modules/autopage/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/autopage/default.nix:13]: They are run via stestr in passthru.

File: pkgs/development/python-modules/cliff/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/cliff/default.nix:45]: Good catch! Thank you!

**Pull request merged**